src/components/Typography/Typography.style.ts

```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components';
+import theme from 'assets/theme';
```

Comment:
should we be using `useTheme()` or something so that we actually benefit from the ThemeProvider infrastructure and can replace the theme dynamically as necessary?

Reply:
We can't use the `useTheme` hook directly on styled components, but we can access the theme variables directly since we have the app wrapped in the styled components theme provider

```tsx
import theme from 'assets/theme';

<ThemeProvider theme={theme}>
  <ScThemeProvider={theme}>
    {/* components here*/}
  <ScThemeProvider/>
</ThemeProvider>
```

in function components, we can use the `useTheme` hook if we need access to theme variables, but for styled components, we should use the direct access to the theme

```ts
// The theme is accessible from ScThemeProvider
const Subtitle1 = styled(Typography).attrs(props => ({ component: 'body1' }))`
  ${props => props.theme.fonts.subtitle1};
`
```

so yes, we should use the theme provider in both cases, but in different ways, will update!